libcloud/compute/drivers/azure_arm.py

```diff
@@ -733,12 +740,14 @@ def destroy_node(self, node, ex_destroy_nic=True, ex_destroy_vhd=True):
 
         # Poll until the node actually goes away (otherwise attempt to delete
         # NIC and VHD will fail with "resource in use" errors).
-        while do_node_polling:
+        retries = ex_retries
+        while do_node_polling and retries > 0:
             try:
                 time.sleep(10)
```

Comment:
I was wondering: in your experience, how much time does it take for a node to go away?

Reply:
A couple of minutes.  The initial delete request is asynchronous, it responds "202 Accepted", which is why we have an idle loop to wait for it to really go away, because we can't delete the other resources (the NIC and VHD) until the leases are released.

Reply:
Sure. Thanks for the answer. In this case, the initial sleep of 10 seconds is acceptable, otherwise it would have made more sense to poll before sleeping.